protocols/kad/tests/client_mode.rs

```diff
@@ -111,6 +111,56 @@ async fn adding_an_external_addresses_activates_server_mode_on_existing_connecti
     }
 }
 
+#[async_std::test]
+async fn set_client_to_server_mode() {
+    let _ = env_logger::try_init();
+
+    let mut client = Swarm::new_ephemeral(MyBehaviour::new);
+    client
+        .behaviour_mut()
+        .kad
+        .set_explicit_mode(Some(Mode::Client));
+
+    let mut server = Swarm::new_ephemeral(MyBehaviour::new);
+
+    server.listen().await;
+    client.connect(&mut server).await;
+
+    let server_peer_id = *server.local_peer_id();
+
+    match libp2p_swarm_test::drive(&mut client, &mut server).await {
+        (
+            [MyBehaviourEvent::Identify(_), MyBehaviourEvent::Identify(_), MyBehaviourEvent::Kad(KademliaEvent::RoutingUpdated { peer, .. })],
+            [MyBehaviourEvent::Identify(_), MyBehaviourEvent::Identify(identify::Event::Received { info, .. })],
+        ) => {
+            assert_eq!(peer, server_peer_id);
+            assert!(info
+                .protocols
+                .iter()
+                .any(|proto| libp2p_kad::PROTOCOL_NAME.ne(proto)))
```

Comment:
```suggestion
            assert!(info
                .protocols
                .iter()
                .all(|proto| libp2p_kad::PROTOCOL_NAME.ne(proto)))
```

As far as I can tell you want to check that all of the protocols are un-equal to the Kademlia protocol, right?

Reply:
That is correct. Even though there is only a single protocol, best logic would be to check them all so it would fail if one is detected.